modules/Bio/EnsEMBL/Variation/DBSQL/LDFeatureContainerAdaptor.pm

```diff
@@ -423,15 +419,33 @@ sub _fetch_by_Slice_VCF {
         my $vcf_file = $vc->_get_vcf_filename_by_chr($slice->seq_region_name);
         throw("ERROR: Can't get VCF file\n") unless $vcf_file;
         push @files, $vcf_file;
-        my $loc_string = sprintf("%s:%i-%i", $slice->seq_region_name, $slice->start, $slice->end);
+        my $chr = $slice->seq_region_name;
+        if ($vc->use_seq_region_synonyms) {
+          my @chr_in_vcf_file = @{$vc->_vcf_parser_obj($vcf_file)->{tabix_file}->seqnames};
+          my @synonyms = ();
+          foreach my $synonym (@{$vc->_get_synonyms_by_chr($chr)}) {
+            push @synonyms, $synonym if (grep {$_ eq $synonym} @chr_in_vcf_file);
+          }
+          $chr = $synonyms[0];
+          if (scalar @synonyms > 1) {
+            warn "use_seq_region_synonyms is set. Found more than one synonym for sequence name $chr\n";
+          } elsif (scalar @synonyms == 0) {
+            warn "use_seq_region_synonyms is set. But didn't find synonym for sequence name $chr\n"; 
+          }
+        } 
+        my $loc_string = sprintf("%s:%i-%i", $chr, $slice->start, $slice->end);
         push @regions, $loc_string;
       }
       my $files_arg = join(',', @files); 
       my $regions_arg = join(',', @regions);
       my $number_of_files = scalar @files;
       my $window_size = $self->max_snp_distance;
-      $cmd = "$bin -f $files_arg -r $regions_arg -s $number_of_files -l $sample_string -w $window_size";
 
+      my $working_dir = cwd();
+      my $return_chdir = chdir $self->temp_path if ($self->temp_path);
+      warn("ERROR: Couldn't change working directory (" . $self->temp_path . "): $!\n") if (!$return_chdir);
```

Comment:
The warn is being reported with no chdir as $return_chdir is only set if ($self->temp_path).
The warn only needs to be given if the chdir was attempted and failed.

```
if ($self->temp_path) {
  my $return_dir = chdir $self->temp_path;
  if (! $return_dir) {
     warn("ERROR: Couldn't change working directory (" . $self->temp_path . "): $!\n")
  }
}
```

```